### ISSUE TITLE
Cache appears to be one step behind

### ISSUE THREAD

**Issue by arcticmatt** [apollographql/apollo-client-devtools#288]
See the attached video to see exactly what I mean.

In words, here's what's happening.

The first time a mutation and is run and my query is refetched, the cache does not update. It only updates on the next refetch. But this means it is one step behind, i.e. the cache (as seen in apollo client devtools) does not reflect the current state. For example, the cache might read
```
▾ROOT_QUERY
decks({}):
0:{"__ref":"Deck:17"}
1:{"__ref":"Deck:18"}
2:{"__ref":"Deck:37"}
3:{"__ref":"Deck:46"}
```

even though only the last deck has already been deleted.

However, when I log in the `merge` function for the field (see code below), I do see that the incoming object is correct (this is also shown in the video).

Not sure if I'm doing something wrong here or if there's a bug.

Here's the code I use to construct the apollo client.

```
  return new ApolloClient({
    ssrMode: typeof window === "undefined",
    link: httpLink,
    cache: new InMemoryCache({
      typePolicies: {
        Query: {
          fields: {
            decks: {
              merge(_ignored, incoming) {
                console.log("ignored", _ignored);
                console.log("incoming", incoming);
                return incoming;
              },
            },
          },
        },
      },
    }),
    defaultOptions,
  });
```

[apollo dev tools.mov.zip](https://github.com/apollographql/apollo-client-devtools/files/5249065/apollo.dev.tools.mov.zip)

**Comment by Marcelo-Figueiredo** [comment]
I'm facing the same problem. Devtools' Apollo cache doesn't reflect the actual Apollo cache, it's one step behind. This is quite problematic because it misleads developers to think their code is wrong.

**Comment by jcreighton** [comment]
We're currently working on a 3.0 version, which is a complete rewrite of the devtools. We won't be working on fixing issues in 2.x so we can focus on getting 3.0 out the door. But if you're able to open a PR for this, I'll definitely take a look!

**Comment by jcreighton** [comment]
We released 2.3.2 so please try that version. Let me know if it resolves this issue.

**Comment by jcreighton** [comment]
Closing this as the latest version should resolve this error. Feel free to re-open if that's not the case!
